services/web/app.py

```diff
@@ -135,15 +140,34 @@ def get(self):
 
 
 class AuthView(views.MethodView):
-    def get(self):
-        print("Begin auth")
-        if request.headers.get("X-Original-Uri", "").startswith("/saml/"):
-            return make_response("Authorized", 200)
-
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.backends = dict()
+        with open(os.environ["BACKENDS_CONFIG_MAP"]) as ifs:
+            backends_list = yaml.safe_load(ifs)
+        for backend in backends_list:
+            if "auth" in backend:
+                self.backends[backend["route"]] = backend["auth"]
+
+    def make_identity_header(self, identity_type, auth_data):
+        header_data = dict(type=identity_type, **auth_data)
+        print(header_data)
+        return base64.encodebytes(json.dumps(header_data).encode("utf8")).replace(b"\n", b"")
+
+    def auth_saml(self, auth):
         if "samlUserdata" in session:
             auth_data = session["samlUserdata"].items()
             print(f"SAML auth_data: {auth_data}")
-            return make_response("Authorized", 200)
+            predicate = auth["saml"]
+            authorized = eval(predicate, dict(user=auth_data))
+            if authorized:
+                resp = make_response("Authorized", 200)
+                resp.headers["X-RH-Identity"] = self.make_identity_header(
+                    "associate", dict(user={k: v if len(v) > 1 else v[0] for k, v in auth_data})
```

Comment:
Just looking at the way we have the IPP and expectations setup for the existing identity [1], should we try to keep the structure similar? Currently we set "System" or "User" for the `type` field. For consistency would we want "Associate"? 

The `user` object we specify currently as only being present for type "User", but we could update the doc to reflect this new type as well, vs a new object.

Assuming we implement some logging similar to the current gateway logs in the future, it would probably be beneficial to set the "auth_type" here as well?

We do normalize the identity in the 3scale gateway (particularly the user data) regardless of the JWT source, or other auth types. Would it make sense to do the same here with the user fields to make service developers' lives easier?

[1] https://docs.google.com/document/d/1PAzJqcUXlxg7t5cX1lsPsQtBPT_95bZbyB9Iiv_ekGM/edit#heading=h.9rvb6r5r6f4n

Reply:
I'm really interested in nailing down the X-RH-Identity spec to something that we can write a schema around. Perhaps we can do this orthogonally to this. But you're right, I could do a better job at conforming to the IPP.